tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+'''
+Utility script to read and write Prometheus Alert rules to file.
+Ensures we can test the Alert rules deployed and for documentation
+purposes.
+'''
+
+import argparse
+import csv
+import ipaddress
+import json
+import pathlib
+import requests
+import subprocess
+import sys
+
+
+def query_prometheus_api(ip, port, route=None):
+    """Read the Prometheus route contents into memory."""
+    if not route:
+        sys.stderr.write("Expected endpoint route but got None")
+        sys.exit(1)
+
+    endpoint = "https://{}:{}/api/prometheus/api/v1/{}".format(ip, port, route)
+    try:
+        response = requests.get(url=endpoint, verify=False)
+    except requests.exceptions.ConnectionError as exc:
+        sys.stderr.write(
+            "Failed to reach endpoint {}: {!s}".format(route, exc)
+        )
+        sys.exit(1)
+
+    if response.status_code != 200:
+        sys.stderr.write(
+            "Expected status code <200> got {}".format(response.status_code)
+        )
+        sys.exit(1)
+
+    return response.json()
+
+
+def write_json_to_file(content, rule_name):
+    filename = '{}.json'.format(rule_name)
+    filepath = pathlib.Path(__file__).parent/filename
+    with open('{}'.format(filepath), 'w', encoding='utf-8') as f:
+        json.dump(content, f, indent=4, sort_keys=True)
+
+
+def write_csv_to_file(content, rule_name):
```

Comment:
CSV? Oh no, the nightmares are coming back...

Reply:
We only use this for automatically generating a table in the docs

Reply:
Yeah I understand why, I'm just a :trollface: 

However, still thinking this intermediate format just to be able to re-use the existing `csv-table` directive is only OK short-term. I'd rather have our own extension build this table directly from the JSON output of Prometheus (single data source, flexible display).